src/network/core.rs

```diff
@@ -74,36 +74,37 @@ impl Core {
         //we have the bridge name but we must iterate for all the available gateways
         for (idx, subnet) in network.subnets.iter().flatten().enumerate() {
             let subnet_mask_cidr = subnet.subnet.prefix_len();
-
-            if let Some(gw) = subnet.gateway {
-                let gw_net = match gw {
-                    IpAddr::V4(gw4) => match ipnet::Ipv4Net::new(gw4, subnet_mask_cidr) {
-                        Ok(dest) => ipnet::IpNet::from(dest),
-                        Err(err) => {
-                            return Err(std::io::Error::new(
-                                std::io::ErrorKind::Other,
-                                format!(
-                                    "failed to parse address {}/{}: {}",
-                                    gw4, subnet_mask_cidr, err
-                                ),
-                            ))
-                        }
-                    },
-                    IpAddr::V6(gw6) => match ipnet::Ipv6Net::new(gw6, subnet_mask_cidr) {
-                        Ok(dest) => ipnet::IpNet::from(dest),
-                        Err(err) => {
-                            return Err(std::io::Error::new(
-                                std::io::ErrorKind::Other,
-                                format!(
-                                    "failed to parse address {}/{}: {}",
-                                    gw6, subnet_mask_cidr, err
-                                ),
-                            ))
-                        }
-                    },
-                };
-
-                gw_ipaddr_vector.push(gw_net)
+            if !network.internal {
+                if let Some(gw) = subnet.gateway {
```

Comment:
with that you no longer assign the gateway ip to the bridge interface.
Actually we might have to do this otherwise an internal container could access services from the host which listen on 0.0.0.0 but how can we support dns with that? @mheon WDYT?

Reply:
We could "cross that bridge" when we have dns ready to implement?

Reply:
If we're disabling forwarding already why go the extra step and also not assign an IP?

Is there an expectation that Docker internal networks are unable to access services on the host? If so, this makes sense. If not, we should leave the IP.

Reply:
I just checked what docker is doing. There are just blocking the access via iptables. You can still connect to interfaces on the hosts.
In this case we definitely still need the gateway ip on the bridge in order to communicate with the dns server.

Reply:
Alright, let's leave the IP on, then.